### PR TITLE
fix:(export): &nbsp; in CSV

### PR DESCRIPTION
**Pull request by Rom1-B**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !30871

When exporting to CSV, some empty data appeared with &nbsp;

Example with the User > Computer Emails column:
image

image

### CODE REVIEW

**Comment by cconard96** [comment]
Does anyone know why the search engine returns `&nbsp;` for some empty results to begin with? In some cases, it is used to add space between an icon and text, but having it on its own doesn't make sense to me.

**Comment by trasher** [comment]
I agree with Curtis; proposed fix seems not correct; no unbreakable space should been present here.

**Comment by Rom1-B** [comment]
It seems deliberate, but the reason escapes me too?

image

**Comment by trasher** [comment]
There are several cases in Search class that return only `&nbsp;` in both `gitveItem()` and `showItem()` methods. I guess they should return empty strings.
I'm not sure since Search class is not easy to understand, but I do not see any reason for that; maybe is this just an old hack kept in place.

**Comment by cconard96** [comment]
The use of &nbsp; has existed for 14 years, since at least the origin of the search class. It's hard to tell what the reasoning was.

**Comment by trasher** [comment]
Indeed it's been present since the origin of the class.

`nbsp` is only for display; it should really not affect the code.
Search results are mainly used in display, which should nowadays not really be affected. It's also probably used from API, in which `nbsp` as empty data makes no sense.

I'd vote to replace with empty string.

File: src/Search.php [hunk 1]
```diff
@@ -8757,7 +8757,7 @@ public static function showItem($type, $value, &$num, $row, $extraparam = '')
                     $value = preg_replace('/' . self::LBBR . '/', '<br>', $value);
                     $value = preg_replace('/' . self::LBHR . '/', '<hr>', $value);
                     $value = '<div class="fup-popup">' . $value . '</div>';
-                    $valTip = "&nbsp;" . Html::showToolTip(
+                    $valTip = '' . Html::showToolTip(
```
Comment (trasher) [inline src/Search.php:8760]: In most of otehr usages in the same class, `showTooltips()` results are separated with a simple space.
```suggestion
                    $valTip = ' ' . Html::showToolTip(
```

Comment (cconard96) [inline src/Search.php:8760]: I think this case is "valid" as it is used as a spacer and shouldn't ever be result in `&nbsp;` being returned on its own.

Comment (trasher) [reply src/Search.php:8760]: (also, this is not a really result data and should not be present in exports or API results I guess)

**Review by trasher — APPROVED** [review]

**Pull request merged**